src/reducers/versions.tsx

```diff
@@ -1458,6 +1481,20 @@ const reducer: Reducer<VersionsState, ActionType<typeof actions>> = (
     }
     case getType(actions.setCurrentVersionId): {
       const { versionId } = action.payload;
+      let { expandedPaths } = state;
+      const version = getVersionInfo(state, versionId);
+
+      if (version && expandedPaths) {
+        const paths = version.entries.map((e) => e.path);
+        expandedPaths = expandedPaths.filter((e) => paths.includes(e));
```

Comment:
This looks right but the `loadVersionInfo` reducer also needs to access this filter. I suggest moving it to a shared function and making unit tests for it as a function.

As an example, `fetchDiff` relies on [`loadVersionInfo`](https://github.com/mozilla/addons-code-manager/blob/9b82a1070efb275eef65a3975fc9732cc27f2100/src/reducers/versions.tsx#L1001) to switch versions and filtering expanded paths would be important at that point. The same thunk dispatches `setCurrentVersionId` earlier on but that happens before the new `version` has loaded.

Reply:
FWIW, it looks that `setCurrentVersionId` is always (?) dispatched again after loading a new version when `_fetchDiff` for preloading next diff. I am fine with adding the filter in `loadVersionInfo` (maybe making it not depend on preloading), but it does not seem to have real benefit except filtering the paths once more.. How do you think about it?

Reply:
We can't depend on preloading since the user can click on any file path out of order. I think it would be safer to introduce a shared function for both reducers to use and it's ok if the paths get filtered twice. You won't have to duplicate any tests for this. I suggest testing it only in the shared function.